---
title: "01_SequenceCollection"
output: html_document
---
# /Users/Yoga/Library/CloudStorage/Dropbox/flag/Full_genomes/Spo_Genomes/protein_20250805.fasta. is the file used for Homolog searches
# protein_20250805.fasta -moved to local folder "/Semester3/Genome" - as it is large and cant be synched with Git

```{bash}
# Bash File for Downloading Genome
chmod +x download_genomes.sh
./download_genomes.sh

# Concat all the protein fasta from 
find GCF_* -name "*protein.faa" -execdir cat '{}' \; > 20250805newprots_allprots.fasta
```
# Search for homology using precalculated HMM profiles
```{bash}
# unzip all the hmm files
find *.hmm.gz -execdir gunzip '{}' \;

# Use hmm to find the distant homologs
# FliF
hmmsearch --cpu 4 -E 0.01 -o ./Data/Homologs_precalculated/FliF/hmm_results_FliF.txt --tblout ./Data/Homologs_precalculated/FliF/FliF_hits_table.txt --domtblout ./Data/Homologs_precalculated/FliF/domains_table.txt ./Data/Precalculated_HMM/YscJ_FliF.hmm ./Data/protein_20250805.fasta

#FliF_C
hmmsearch --cpu 4 -E 0.01 -o ./Data/Homologs_precalculated/FliF_C/hmm_results_FliF_C.txt --tblout ./Data/Homologs_precalculated/FliF_C/FliF_C_hits_table.txt --domtblout ./Data/Homologs_precalculated/FliF_C/domains_table.txt ./Data/Precalculated_HMM/YscJ_FliF_C.hmm ./Data/protein_20250805.fasta

#FlhB
hmmsearch --cpu 4 -E 0.01 -o ./Data/Homologs_precalculated/FlhB_SpaS/hmm_results_FlhB_SpaS.txt --tblout ./Data/Homologs_precalculated/FlhB_SpaS/FlhB_SpaS_hits_table.txt --domtblout ./Data/Homologs_precalculated/FlhB_SpaS/domains_table.txt ./Data/Precalculated_HMM/FlhB_SpaS_HrpN_YscU.hmm ./Data/protein_20250805.fasta

#SpoIIIAH
hmmsearch --cpu 4 -E 0.01 -o ./Data/Homologs_precalculated/SpoIIIAH/hmm_results_SpoIIIAH.txt --tblout ./Data/Homologs_precalculated/SpoIIIAH/SpoIIIAH_hits_table.txt --domtblout ./Data/Homologs_precalculated/SpoIIIAH/domains_table.txt ./Data/Precalculated_HMM/SpoIIIAH.hmm ./Data/protein_20250805.fasta
```

# Extract fasta file based on HMM hits
```{bash}
./Data/Homologs_precalculated/FliF/FliF_hits_table.txt
# FliF
grep -v '^#' ./Data/Homologs_precalculated/FliF/FliF_hits_table.txt | awk '{print $1}' | sort -u > ./Data/Homologs_precalculated/FliF/FliF_hit_ids.txt
seqkit grep -f ./Data/Homologs_precalculated/FliF/FliF_hit_ids.txt ./Data/protein_20250805.fasta > ./Data/Homologs_precalculated/FliF/FliF_matched_hits.fasta

# FliF_C
grep -v '^#' ./Data/Homologs_precalculated/FliF_C/FliF_C_hits_table.txt | awk '{print $1}' | sort -u > ./Data/Homologs_precalculated/FliF_C/FliF_C_hit_ids.txt
seqkit grep -f ./Data/Homologs_precalculated/FliF_C/FliF_C_hit_ids.txt ./Data/protein_20250805.fasta > ./Data/Homologs_precalculated/FliF_C/FliF_C_matched_hits.fasta

# SpoIIIAH
grep -v '^#' ./Data/Homologs_precalculated/SpoIIIAH/SpoIIIAH_hits_table.txt | awk '{print $1}' | sort -u > ./Data/Homologs_precalculated/SpoIIIAH/SpoIIIAH_hit_ids.txt
seqkit grep -f ./Data/Homologs_precalculated/SpoIIIAH/SpoIIIAH_hit_ids.txt ./Data/protein_20250805.fasta > ./Data/Homologs_precalculated/SpoIIIAH/SpoIIIAH_matched_hits.fasta

# FlhB
grep -v '^#' ./Data/Homologs_precalculated/FlhB_SpaS/FlhB_SpaS_hits_table.txt | awk '{print $1}' | sort -u > ./Data/Homologs_precalculated/FlhB_SpaS/FlhB_SpaS_hit_ids.txt
seqkit grep -f ./Data/Homologs_precalculated/FlhB_SpaS/FlhB_SpaS_hit_ids.txt ./Data/protein_20250805.fasta > ./Data/Homologs_precalculated/FlhB_SpaS/FlhB_SpaS_matched_hits.fasta

## Concat all fasta files
cat ./Data/Homologs_precalculated/FliF/FliF_matched_hits.fasta ./Data/Homologs_precalculated/FliF_C/FliF_C_matched_hits.fasta ./Data/Homologs_precalculated/SpoIIIAH/SpoIIIAH_matched_hits.fasta ./Data/Homologs_precalculated/FlhB_SpaS/FlhB_SpaS_matched_hits.fasta > ./Data/Homologs_precalculated/homologs.fasta
```

# Preliminary Analysis
```{bash}
## Removing duplicate sequences
seqkit rmdup -s < homologs.fasta > homologs_duplicate_removed_seqkit.fasta
```

# HMM Building and analysis
```{bash}
# Copying start sequence fasta
fasta.sh 

# Building sto (using esl-reformat) from aligned fasta (using Mafft)
fasta2sto.sh

## Conversion of sto to hmm prolfile
sto2hmm.sh (hmmbuild aligned.hmm aligned.sto)

## hmmsearch using the profiles

hmm2fasta.sh (hmmsearch fron /Users/Yoga/Library/CloudStorage/Dropbox/flag/Full_genomes/Spo_Genomes/protein_20250805.fasta)

## Extract fasta file based on HMM hits
hmm2homologs.sh

## Concating fasta files
concat_homologs.sh

## duplicate removal using seqkit 

```

```{bash}
seqkit rmdup -s < homologs.fasta > homologs_duplicate_removed_seqkit.fasta
```


