---
title: "Visualisation__Preliminary_Pecalculated"
output: html_document
---
## Loading libraries
```{r}
library(ape)
library(msa)
library(seqinr)
library(BioGeoBEARS)
library(rvest) # for html_table()
library(phytools) # for midpoint.root
library(openxlsx) # for read.xlsx
library(ape)
library(phytools)

```


## Renaming Homolog file with accession number
```{r}
getwd()
xlsfn = "Data/03_Homologs_Jackhmmer/Homologs_renamed.xlsx"
xls = openxlsx::read.xlsx(xlsxFile=xlsfn, sheet=1, startRow=1, colNames=TRUE)
fasta_file <- read.fasta (file = "Data/03_Homologs_Jackhmmer/homologs_duplicate_removed_seqkit.fasta")
newnames = as.array(xls$protein)
names(fasta_file) <- sapply(names(fasta_file), function(name) {
  fasta_ids <- strsplit(name, split = "\\.")[[1]][1]
   for (j in 1:length(newnames)){
    newnames_accession <- strsplit(newnames[j], split = "\\.")[[1]][1]
    if (fasta_ids == newnames_accession) {
      return(newnames[j])
   }
   }
})
write.fasta(sequences = fasta_file, names=names(fasta_file), file.out = "Data/03_Homologs_Jackhmmer/homologs_duplicate_removed_seqkit_renamed.fasta")
```

## Renaming Aligned File with Accession number Clustal_Nj
```{r}
xlsfn = "Data/03_Homologs_Jackhmmer/Homologs_renamed.xlsx"
xls = openxlsx::read.xlsx(xlsxFile=xlsfn, sheet=1, startRow=1, colNames=TRUE)
fasta_file <- read.fasta (file = "Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/input.fa.final_tree.used_alg.fa")
newnames = as.array(xls$protein)
names(fasta_file) <- sapply(names(fasta_file), function(name) {
  fasta_ids <- strsplit(name, split = "\\.")[[1]][1]
   for (j in 1:length(newnames)){
    newnames_accession <- strsplit(newnames[j], split = "\\.")[[1]][1]
    if (fasta_ids == newnames_accession) {
      return(newnames[j])
   }
   }
})
write.fasta(sequences = fasta_file, names=names(fasta_file), file.out = "Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/input.fa.final_tree.used_alg_renamned.fa")

```

## Renaming tree file
```{r}
source("~/GitHub/str2phy/Rsrc/seqnames_v1.R")
library(tidyverse)

tree_fn = "Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/outTree_midpointRooted.nwk"
tr = ape::read.tree(tree_fn)
xlsfn = "Data/03_Homologs_Jackhmmer/Homologs_renamed.xlsx"
xls = openxlsx::read.xlsx(xlsxFile=xlsfn, sheet=1, startRow=1, colNames=TRUE)

tipnames <- ifelse(
  grepl("\\.", tr$tip.label),
  substr(tr$tip.label, 1, (regexpr("\\.", tr$tip.label)+1)),
  tr$tip.label
)

tr$tip.label = tipnames
gidskey = xls$AccessionNumber
newmatch = fixnames(xls$protein)
tr_renamed = fix_tipnames_gid(gids=gidskey, newmatch=newmatch, tr=tr)

ape::write.tree(tr_renamed, "Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/outTree_midpointRooted_renamed.nwk")
```

# Reordering Alignment File based on tree
```{r}
tr_renamed <- ape::read.tree("Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/outTree_midpointRooted_renamed.nwk")
desired_order <- tr_renamed$tip.label[order(-(get("last_plot.phylo", envir = .PlotPhyloEnv)$yy[1:Ntip(tr_renamed)]))]
fasta_file <- read.fasta (file = "Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/input.fa.final_tree.used_alg_renamned.fa")
fasta_ids <-names(fasta_file)
tip_ids <- desired_order
reordered_fasta <- fasta_file[(match(tip_ids, fasta_ids))]
write.fasta(sequences = reordered_fasta, names=desired_order, file.out = "Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/input.fa.final_tree.used_alg_renamned_reordered.fa")
```


## Clustal_Nj
```{r}
clustal_tree <- read.tree("Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/outTree_midpointRooted_renamed.nwk")
pdf("Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/Tree.pdf", width = 8, height = 8)
plot(clustal_tree, edge.width = 0.05, label.offset = 3, cex = 0.05)
dev.off()

```
### To identify the issues in fig tree  - nodes cannot be empty error
```{r}
library(ape)

# Read the tree
tree_fn <- "Data/03_Homologs_Jackhmmer/Seqkit_duplicate_removed/Clustal_Nj/outTree_midpointRooted_renamed.nwk"
tr <- ape::read.tree(tree_fn)

# Inspect tip labels
tip_labels <- tr$tip.label
cat("First 10 tip labels:\n")
print(head(tip_labels, 10))

# Inspect internal node labels
node_labels <- tr$node.label
cat("First 10 internal node labels:\n")
print(head(node_labels, 10))

# Check for empty tip labels
empty_tips <- tip_labels[tip_labels == "" | is.na(tip_labels)]
cat("Empty tip labels:\n")
print(empty_tips)

# Check for empty node labels
empty_nodes <- node_labels[node_labels == "" | is.na(node_labels)]
cat("Empty internal node labels:\n")
print(empty_nodes)

# Check for duplicate tips
dup_tips <- tip_labels[duplicated(tip_labels)]
cat("Duplicate tip labels:\n")
print(dup_tips)

# Check for illegal characters in tip labels
illegal_chars <- grep("[:()\\[\\],; ]", tip_labels, value = TRUE)
cat("Tip labels with illegal characters:\n")
print(illegal_chars)

```


